src/box/raft.c

```diff
@@ -201,6 +201,14 @@ box_raft_on_update_f(struct trigger *trigger, void *event)
 	 */
 	box_update_ro_summary();
 	box_broadcast_election();
+	/*
+	 * Once the node becomes read-only due to new term, it should stop
+	 * finalizing existing synchronous transactions so that it doesn't
+	 * trigger split-brain with a new leader which will soon emerge.
+	 */
+	if (raft->volatile_term > txn_limbo.promote_greatest_term) {
+		txn_limbo_fence(&txn_limbo);
+	}
```

Comment:
Nit: you don't need `{}` for one-line ifs here and in other places in other commits. Up to you anyway.

Reply:
Sure. Fixed.